### PR TITLE
Feature/categories

### PR DESCRIPTION
**Pull request by mateomm22**
### Categories ###

Creation of the categories container.
Fetch the categories and render them inside the container using redux and dispatching actions to do so.

_Changed the way stylus is being compiled in order to make it more efficient_

### CODE REVIEW

File: src/containers/categories.js [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
 import CatGroup from '../components/category/catGroup';
 
 class Categories extends Component {
+
+  componentDidMount() {
+    this.props.getTypes();
+    this.props.getAlc();
+  }
+
+  filterDrinks(idCat,catName) {
```
Comment (jorsebsant) [inline src/containers/categories.js]: if this will be merged it should have console.logs, are you sure you are running linter to avoid this stuff?

Comment (mateomm22) [reply src/containers/categories.js]: I'm not running linter yet, i'll use ESLint after this PR is that ok?

Comment (mateomm22) [reply src/containers/categories.js]: Should I leave just the empty function without the console.logs? or delete the entire function and its reference inside the component?

File: src/assets/layout/home.styl [hunk 2]
```diff
@@ -1,3 +1,22 @@
 .home
   text-align center
-  
\ No newline at end of file
+  padding-top 50px
+
+.searcher
+  max-width 450px
```
Comment (jorsebsant) [inline src/assets/layout/home.styl]: one of the best things to do is to order all the css properties alphabetically

Comment (jorsebsant) [reply src/assets/layout/home.styl]: if you do, do it with all the css in this pr and after on the rest of the files of the project

File: src/store/reducers/alcohol.js [hunk 3]
```diff
@@ -1,5 +1,30 @@
-import { initialState } from '../data/alcohol';
+import * as actions from '../actions/actionTypes';
 
-const alcoholReducer = (state = initialState, action) => state;
+const setLevels = (state, action) => {
+  
+  let results = [];
+  action.drinks.map((type, key) => {
```
Comment (jorsebsant) [inline src/store/reducers/alcohol.js]: are you sure you are using the map function in the right way? it looks like you were using a very old way to solve this

File: src/containers/results.js [hunk 4]
```diff
@@ -1,13 +1,63 @@
...
 class Results extends Component {
-  render (){
+  
+  componentWillUnmount(){
+    this.props.clearState();
+  }
+  
+  render() {
+    
+    let drinks = this.props.results.map((drink, key) => {
```
Comment (jorsebsant) [inline src/containers/results.js]: when does this variable change to be considered as a "let"

File: src/containers/results.js [hunk 5]
```diff
@@ -1,13 +1,63 @@
...
+      return (
```
Comment (jorsebsant) [inline src/containers/results.js]: you can use implicit return for this, we are not using any logic inside this function

File: src/misc/layout.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
-const Layout = (props) => {
+const layout = (props) => {
```
Comment (jorsebsant) [inline src/misc/layout.js]: for convention this should be an uppercase thing, because at the end of the day it is a component

**Review by jorsebsant — CHANGES_REQUESTED** [review]
solve this comments first, meanwhile i will be taking a deeper look.

File: debug.log [hunk 7]
```diff
@@ -0,0 +1 @@
+[0220/151449.180:ERROR:crash_report_database_win.cc(428)] unexpected header
```
Comment (jorsebsant) [inline debug.log:1]: this is supposed to be ignored, so maybe this was already tracked, tri to untrack it or delete from the repo

File: src/components/category/catGroup.js [hunk 8]
```diff
@@ -1,13 +1,17 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+
 import CatItem from './catItem';
 
 
 const catGroup = (props) => {
   let items = props.items.map(item => {
+    if(!item.name) return false;
```
Comment (jorsebsant) [inline src/components/category/catGroup.js]:
```suggestion
    if(item.name)
```

would be this better?

Comment (jorsebsant) [reply src/components/category/catGroup.js]: just return the ones with  name instead of returning false, maybe im wrong, test it first.

File: src/components/category/catGroup.js [hunk 9]
```diff
@@ -1,13 +1,17 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+
 import CatItem from './catItem';
 
 
 const catGroup = (props) => {
   let items = props.items.map(item => {
```
Comment (jorsebsant) [inline src/components/category/catGroup.js]: where does this change to be considered as a let?

File: src/components/category/catGroup.js [hunk 10]
```diff
@@ -1,13 +1,17 @@
...
+    if(!item.name) return false;
     return (
       <CatItem 
         key={item.id}
-        name={item.name} />
+        name={item.name}
+        clicked={() => props.itemClicked(props.idCat, item.name)} />
```
Comment (jorsebsant) [inline src/components/category/catGroup.js]: just checking, props.idCat is not the same as item.id? i'm not sure, just checking

File: src/components/results/resultItem.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const resultItem = (props) => {
```
Comment (jorsebsant) [inline src/components/results/resultItem.js]: use implicit return for this, we have no logic inside of it to use the brackets

File: src/store/actions/drinkCategories.js [hunk 12]
```diff
@@ -14,9 +14,6 @@ export const fetchDrinkTypes = () => {
       .get('https://www.thecocktaildb.com/api/json/v1/1/list.php?c=list')
       .then(response => {
         dispatch(setFetchedTypes(actions.GET_DRINK_TYPES, response.data.drinks));
-      })
-      .catch(error => {
-        dispatch(console.log(error));
```
Comment (jorsebsant) [inline src/store/actions/drinkCategories.js:19]: why did you delete the "catch" here and below, i know it have a console.log but then lets handle this in another way but the catch are pretty much necessary, if you only deleted this because of the console.log, please let me know

File: src/store/reducers/alcohol.js [hunk 13]
```diff
@@ -1,5 +1,29 @@
...
 
-const alcoholReducer = (state = initialState, action) => state;
+const setLevels = (state, action) => {
+  
+  const results = action.drinks.map((type, key) => {
+    let id = key;
+    let name = type.strAlcoholic;
+    let typeInfo = {
+      id: id,
+      name: name
```
Comment (jorsebsant) [inline src/store/reducers/alcohol.js]: when the key and the variable assigned to it have the same name there is no need to duplicate them so, this can be
 {id, name}

Comment (jorsebsant) [reply src/store/reducers/alcohol.js]: or if you prefer you can avoid the creation of the 2 variables above and just calling them inside the object as you prefer

File: src/store/reducers/drinkTypes.js [hunk 14]
```diff
@@ -1,5 +1,29 @@
-import { initialState } from '../data/drinkTypes';
+import * as actions from '../actions/actionTypes';
 
-const drinkTypesReducer = (state = initialState, action) => state;
+const setTypes = (state, action) => {
+  
+  const results = action.drinks.map((type, key) => {
```
Comment (jorsebsant) [inline src/store/reducers/drinkTypes.js]: dont you thing this is pretty similar to the one that is being used in alcohol.js? DRY

File: src/store/reducers/drinks.js [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+const initialState = {
+  loading: false,
+  drinks: []
+}
+
+const setDrinks = (state, action) => {
+  
+  const results = action.drinks.map(drink => {
+    let id = drink.idDrink;
```
Comment (jorsebsant) [inline src/store/reducers/drinks.js]: destructuring in this 3 variables this will safe the other 2 lines of code

Comment (jorsebsant) [reply src/store/reducers/drinks.js]: also i will hit you in the head hahaha, please, you are using too much let when the let cases are pretty low compared to const, this is related to performance. ONLY USE LET IF IT WILL CHANGE, IF NOT USE CONST. take that in mind for every thing you do and all the files in this pr. if you have some confussion between when and when not to use it, consider reviewing this topic about block variables.

File: src/components/category/catGroup.js (hunk 10 above)
Comment (mateomm22) [reply src/components/category/catGroup.js]: They're different, the **props.idCat** is the parent's id (dtrink type, alcohol) and the **item.id** is the id of every children inside that category.

File: src/store/actions/drinkCategories.js [hunk 16]
```diff
@@ -1,35 +1,23 @@
...
-      .then(response => {
-        dispatch(setFetchedTypes(actions.GET_ALC_LEVEL, response.data.drinks));
-      })
-      .catch(error => {
-        dispatch(console.log(error));
-      });
-  };
+export const fetchAlcLevel = () => (dispatch) => {
+  axios
+    .get('https://www.thecocktaildb.com/api/json/v1/1/list.php?a=list')
```
Comment (jorsebsant) [inline src/store/actions/drinkCategories.js:19]: This URL i being used in different places, let's create a constants file for this

**Review by jorsebsant — APPROVED** [review]
